goldstone/core/tasks.py

```diff
@@ -165,45 +164,59 @@ def check_for_pending_alerts():
         obj.last_end = end
         obj.save()
+        if num_hits > 0:
+            saved_alert_dict[obj] = num_hits
 
-        if response.hits.total > 0:
-            # We have a non-zero match for pending alerts
-            # Go ahead and generate an instance of the alert object here.
-            # We can directly call the producer class to send an email
-
-            msg_dict = obj.build_alert_template(hits=response.hits.total)
-
-            # For this scheduled celery task, pick up the message template
-            # from the query object and pass it along. For all other
-            # cases, user is allowed to send custom msg_title and msg_body
-            alert_obj = Alert(query=obj, msg_title=msg_dict['title'],
-                              msg_body=msg_dict['body'])
-            alert_obj.save()
-
-            # Filter by fk = AlertSearch obj
-            # dont throw an exception from this loop and keep retrying
-            # till all the producers in the list are exhausted
-            producer_rv_list = list()
-            for producer in EmailProducer.objects.filter(query=obj):
-                try:
-                    producer_ret = producer.send(alert_obj)
-                    ret_dict = {producer.query.name: producer_ret}
-                except Exception as e:
-                    ret_dict = {producer.query.name: e}
-                    # Uncomment the lines below if we ever want to mark
-                    # this task to be in retry state. For now, we don't
-                    # mind that this task is marked success/failure.
-                    # check_for_pending_alerts.retry(throw=False)
-                    # raise RetryTaskError(None, None)
-                producer_rv_list.append(ret_dict)
-
-            return producer_rv_list
+    for obj in sql_alerts:
+        s, start, end = obj.search_recent()
+        num_hits = len(s)
+        obj.last_start = start
+        obj.last_end = end
+        obj.save()
+        if num_hits > 0:
+            saved_alert_dict[obj] = num_hits
+
+    for obj,num_hits in saved_alert_dict:
+
+        # We have a non-zero match for pending alerts
+        # Go ahead and generate an instance of the alert object here.
+        # We can directly call the producer class to send an email
+
+        msg_dict = obj.build_alert_template(hits=num_hits)
+
+        # For this scheduled celery task, pick up the message template
+        # from the query object and pass it along. For all other
+        # cases, user is allowed to send custom msg_title and msg_body
+        alert_obj = Alert(query=obj, msg_title=msg_dict['title'],
+                          msg_body=msg_dict['body'])
+        alert_obj.save()
+
+        # Filter by fk = AlertSearch obj
+        # dont throw an exception from this loop and keep retrying
+        # till all the producers in the list are exhausted
+        producer_rv_list = list()
+        for producer in EmailProducer.objects.filter(query=obj):
```

Comment:
yeah, this was why we were using django-polymorphic for other stuff.  defer until we have additional producer types and we can evaluate whether we want to apply it here.